Isodose/qSlicerIsodoseModuleWidget.cxx

```diff
@@ -800,26 +973,34 @@ void qSlicerIsodoseModuleWidget::updateScalarBarsFromSelectedColorTable()
 
   int newNumberOfColors = selectedColorNode->GetNumberOfColors();
 
+  vtkMRMLIsodoseNode::DoseUnitsType doseUnits = paramNode->GetDoseUnits();
+  bool relativeRepresentation = paramNode->GetRelativeRepresentationFlag();
+  
   // Get dose unit name and assemble scalar bar title
-  std::string doseUnitName("");
-  vtkIdType doseShItemID = shNode->GetItemByDataNode(doseVolumeNode);
-  if (doseShItemID != vtkMRMLSubjectHierarchyNode::INVALID_ITEM_ID)
+  std::string scalarBarTitle("Isolevels");
+  switch (doseUnits)
   {
-    doseUnitName = shNode->GetAttributeFromItemAncestor(
-      doseShItemID, vtkSlicerRtCommon::DICOMRTIMPORT_DOSE_UNIT_NAME_ATTRIBUTE_NAME, vtkMRMLSubjectHierarchyConstants::GetDICOMLevelStudy());
-  }
-  std::string scalarBarTitle("Dose");
-  if (!doseUnitName.empty())
-  {
-    scalarBarTitle += " (" + doseUnitName + ")";
+  case vtkMRMLIsodoseNode::DoseUnitsType::GY:
+    {
+      const char* str = relativeRepresentation ? " (%)" : " (Gy)";

```

Comment:
Wrong indentation. Please fix here and below

Reply:
Fixed.